### PR TITLE
adds sns ability to send http(s) request 

### PR DESCRIPTION
**Pull request by kcajmagic**

### CODE REVIEW

**Comment by p4tin** [comment]
@kcajmagic seems your PR is failing in TravisCI can you check it out?

**Comment by p4tin** [comment]
@kcajmagic sorry to do this to you again but your need to rebase your PR.

**Comment by kcajmagic** [comment]
@p4tin thats fine, part of the deal with coding. 

What do you think about the code additions. Want me to change anything?

**Comment by kcajmagic** [comment]
@p4tin could you please review this PR.

**Comment by p4tin** [comment]
@kcajmagic - Sorry real busy at work and I want to review properly - I will get it done this weekend.

File: app/gosns/gosns.go [hunk 1]
```diff
@@ -26,6 +47,45 @@ func init() {
 	app.SnsErrors["TopicExists"] = err3
 	err4 := app.SnsErrorType{HttpError: http.StatusBadRequest, Type: "InvalidParameter", Code: "AWS.SimpleNotificationService.ValidationError", Message: "The input fails to satisfy the constraints specified by an AWS service."}
 	app.SnsErrors["ValidationError"] = err4
+	PrivateKEY, PemKEY, _ = createPemFile()
```
Comment (p4tin) [inline app/gosns/gosns.go]: you return errors but do not check them?

File: app/gosns/gosns.go [hunk 2]
```diff
@@ -301,47 +471,19 @@ func Publish(w http.ResponseWriter, req *http.Request) {
 
 	_, ok := app.SyncTopics.Topics[topicName]
 	if ok {
-		log.Println("Publish to Topic:", topicName)
+		log.WithFields(log.Fields{
```
Comment (p4tin) [inline app/gosns/gosns.go]: 👍

**Pull request merged**